File: Data Structures & Algorithms/reverse-bits/submission-1.py
class Solution:
    def reverseBits(self, n: int) -> int:
        if n == 0:
            return 0
        queue = []
        while n > 0:
            if n & 1 == 1:
                queue.append(1)
            else:
                queue.append(0)
            n = n >> 1

        out = queue.pop(0)
        zero_pad = 31 - len(queue)
        while queue:
            out = out << 1
            if queue.pop(0) == 1:
                out = out | 1
        out = out << zero_pad
        return out

File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();
        for(int i = 0; i < s.length(); i++) {
            char cur = s.charAt(i);
            if(cur == '(' || cur == '{' || cur == '[') {
                stk.push(cur);
            }
            else {
                if(stk.empty()) { return false; }
                if((cur == ')' && stk.peek() == '(')
                || (cur == ']' && stk.peek() == '[')
                || (cur == '}' && stk.peek() == '{')) { 
                    stk.pop(); 
                    }
                else return false;
            }
        }
        if(stk.empty()) { return true; }
        return false;
    }
}
